snippets/base/forms.py

```diff
@@ -119,6 +119,24 @@ class BaseSnippetAdminForm(forms.ModelForm):
 class SnippetAdminForm(BaseSnippetAdminForm):
     template = forms.ModelChoiceField(queryset=SnippetTemplate.objects.exclude(hidden=True),
                                       widget=TemplateSelect)
+    client_option_has_fxaccount = forms.ChoiceField(
+        label='Firefox Account',
+        choices=(('any', 'Show to all users'),
+                 ('yes', 'Show only to users with an enabled Firefox Account'),
+                 ('no', 'Show only to users without an enabled Firefox Account')))
+    client_option_has_testpilot = forms.ChoiceField(
+        label='Test Pilot',
+        choices=(('any', 'Show to all users'),
+                 ('yes', 'Show only to users with TestPilot'),
+                 ('no', 'Show only to users without TestPilot')))
+
+    def __init__(self, *args, **kwargs):
+        super(SnippetAdminForm, self).__init__(*args, **kwargs)
+
+        for key in self.fields.keys():
+            if key.startswith('client_option_'):
+                self.fields[key].initial = self.instance.client_options.get(
```

Comment:
Probably you need to check here that the `self.instance.client_options` exists and it's not `None`.